### ISSUE TITLE
Error on build

### ISSUE THREAD

**Issue by paulwhitehead74** [johrstrom/jmeter-prometheus-plugin#52]
Looks to be an error when building master mvn clean package:
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.533 s <<< FAILURE! - in com.github.johrstrom.config.gui.ConfigGuiTest
[ERROR] simpleTest(com.github.johrstrom.config.gui.ConfigGuiTest)  Time elapsed: 0.533 s  <<< ERROR!
java.lang.NullPointerException

**Comment by johrstrom** [comment]
Thanks for the ticket, though I cannot reproduce. I even tried cloning into a brand new directory and building in that new directory, can you try the same?

Also, I now offer jars in the 'release' section, so if you don't necessarily need to build from scratch, you can take a release from there. 

Can you check your `git status` and see if there's something else there?  Failing all of those spot checks, you'll probably have to upload a `mvn clean test -X` output.  If you look at that test, it's really just a spot check itself to be sure one can [instantiate that type](https://github.com/johrstrom/jmeter-prometheus-plugin/blob/master/src/test/java/com/github/johrstrom/config/gui/ConfigGuiTest.java). 

I also tried java 7 (won't even compile), removing all the jmeter 3.2 jars and fetching them again from maven central (that didn't change anything).

**Comment by newbie-qa** [comment]
Hi, I just cloned and tried to build it but i also get the same error mentioned by paul. Could you please take a look ? Many thanks! :)

**Comment by johrstrom** [comment]
Thanks for commenting. Although I can still not reproduce, I did check travis cl (I'd forgotten it was building there) and it's failing there too.  Not sure why, but I'm looking into it.

**Comment by johrstrom** [comment]
I should have actually waited to comment because I had the commit already and was watching the travis cl build while I typed.

This should be fixed on the master branch now, can you both try?  Laughably, all I had to do was set the jmeter locale. 

https://github.com/johrstrom/jmeter-prometheus-plugin/commit/e19aff1803ad6c03f6ccd2fa29a5292bedd4dd7d

**Comment by newbie-qa** [comment]
Thanks for the update @johrstrom :)
Itried to rebuild the new version and i got a new error:

```
unregistering test_summary_rtime
unregistering test_summary_connect_time
[ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.582 s <<< FAILURE! - in com.github.johrstrom.listener.PrometheusListenerTest
[ERROR] canReadJMX(com.github.johrstrom.listener.PrometheusListenerTest)  Time elapsed: 0.038 s  <<< ERROR!
com.thoughtworks.xstream.io.StreamException:
        at com.thoughtworks.xstream.io.xml.XppReader.pullNextEvent(XppReader.java:124)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.readRealEvent(AbstractPullReader.java:148)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.readEvent(AbstractPullReader.java:141)

```

**Comment by johrstrom** [comment]
Has it been modified? This one neither me or a refresh or [travis cl](https://travis-ci.org/johrstrom/jmeter-prometheus-plugin) and replicate the error. Can you check with a `git status` if it's different from what you checked out?

**Comment by xbif** [comment]
I also have error while build:
Loading file: src\test\resources\simple_prometheus_example.jmx
setting new collectors. size is: 0
[ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.085 s <<< FAILURE! - in com.github.johrstrom.listener.PrometheusListenerTest
[ERROR] canReadJMX(com.github.johrstrom.listener.PrometheusListenerTest)  Time elapsed: 0.018 s  <<< ERROR!
com.thoughtworks.xstream.io.StreamException:
        at com.thoughtworks.xstream.io.xml.XppReader.pullNextEvent(XppReader.java:124)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.readRealEvent(AbstractPullReader.java:148)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.readEvent(AbstractPullReader.java:141)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.move(AbstractPullReader.java:118)
        at com.thoughtworks.xstream.io.xml.AbstractPullReader.moveDown(AbstractPullReader.java:103)
        at com.thoughtworks.xstream.io.xml.XppReader.<init>(XppReader.java:63)
        at com.thoughtworks.xstream.io.xml.AbstractXppDriver.createReader(AbstractXppDriver.java:54)
        at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1085)
        at org.apache.jmeter.save.SaveService.readTree(SaveService.java:442)
        at org.apache.jmeter.save.SaveService.loadTree(SaveService.java:425)
        at com.github.johrstrom.listener.PrometheusListenerTest.canReadJMX(PrometheusListenerTest.java:68)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
        at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:410)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.xmlpull.v1.XmlPullParserException: only whitespace content allowed before start tag and not . (position: START_DOCUMENT seen .... @1:1)
        at org.xmlpull.mxp1.MXParser.parseProlog(MXParser.java:1519)
        at org.xmlpull.mxp1.MXParser.nextImpl(MXParser.java:1395)
        at org.xmlpull.mxp1.MXParser.next(MXParser.java:1093)
        at com.thoughtworks.xstream.io.xml.XppReader.pullNextEvent(XppReader.java:109)
        ... 28 more

**Comment by johrstrom** [comment]
It seems there's an issue on windows with the 2 files and one of them being symlinked. I may have the correction in the branch below.  I'm closing this ticket so that we can follow this issue in #53 because the original error this ticket was opened with is closed. Thanks all!

**Comment by johrstrom** [comment]
This issue seems to have been fixed in #53. Thanks all for the ticket and comments. I'm pulling the fix into master now.
